ui/timeline/src/commonMain/kotlin/social/androiddev/timeline/navigation/TimelineViewModel.kt

```diff
@@ -0,0 +1,41 @@
+package social.androiddev.timeline.navigation
+
+import com.arkivanov.essenty.instancekeeper.InstanceKeeper
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import org.mobilenativefoundation.store.store5.ResponseOrigin
+import org.mobilenativefoundation.store.store5.StoreResponse
+import social.androiddev.domain.timeline.FeedType
+import social.androiddev.domain.timeline.HomeTimelineRepository
+import social.androiddev.domain.timeline.model.StatusUI
+import kotlin.coroutines.CoroutineContext
+
+class TimelineViewModel(
+    private val mainContext: CoroutineContext,
+    private val homeTimelineRepository: HomeTimelineRepository
+) : InstanceKeeper.Instance {
+    private val scope = CoroutineScope(mainContext + SupervisorJob())
+    private val _state = MutableStateFlow<StoreResponse<List<StatusUI>>>(StoreResponse.Loading(ResponseOrigin.SourceOfTruth))
+    val state: StateFlow<StoreResponse<List<StatusUI>>> = _state.asStateFlow()
+    init {
+        scope.launch {
+           try {
+               homeTimelineRepository.read(FeedType.Home, refresh = true).collect{
+                   _state.value = it
+               }
+           } catch (e:Exception){
```

Comment:
Also I'd suggest using `runCatchingIgnoreCancelled` here instead and probably you will/could map the error case to a different state, and will bubble up the special CancellationException

Reply:
removed entirely, store returns data/error/loading now without ever throwing an exception